{
  "active_lock_reason": null,
  "assignee": null,
  "assignees": [],
  "author_association": "MEMBER",
  "body": "I got the following failure in the unit test of k8s.io/kubernetes/pkg/controller/volume/persistentvolume.\n\n```\n==================\nWARNING: DATA RACE\nRead by goroutine 156:\n  k8s.io/kubernetes/pkg/controller/volume/persistentvolume.(*PersistentVolumeController).Stop()\n      /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/volume/persistentvolume/controller_base.go:471 +0xc0\n  k8s.io/kubernetes/pkg/controller/volume/persistentvolume.TestControllerSync()\n      /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/volume/persistentvolume/controller_test.go:204 +0x39b2\n  testing.tRunner()\n      /usr/local/go/src/testing/testing.go:473 +0xdc\n\nPrevious write by goroutine 173:\n  k8s.io/kubernetes/pkg/controller/volume/persistentvolume.(*PersistentVolumeController).Run()\n      /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/volume/persistentvolume/controller_base.go:461 +0x2d4\n\nGoroutine 156 (running) created at:\n  testing.RunTests()\n      /usr/local/go/src/testing/testing.go:582 +0xae2\n  testing.(*M).Run()\n      /usr/local/go/src/testing/testing.go:515 +0x11d\n  main.main()\n      k8s.io/kubernetes/pkg/controller/volume/persistentvolume/_test/_testmain.go:90 +0x210\n\nGoroutine 173 (finished) created at:\n  k8s.io/kubernetes/pkg/controller/volume/persistentvolume.TestControllerSync()\n      /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/volume/persistentvolume/controller_test.go:178 +0x32ad\n  testing.tRunner()\n      /usr/local/go/src/testing/testing.go:473 +0xdc\n```\n\nhttps://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/30913/kubernetes-pull-test-unit-integration/40492/\n",
  "closed_at": "2016-08-21T18:36:46Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/980082?v=4",
    "events_url": "https://api.github.com/users/liggitt/events{/privacy}",
    "followers_url": "https://api.github.com/users/liggitt/followers",
    "following_url": "https://api.github.com/users/liggitt/following{/other_user}",
    "gists_url": "https://api.github.com/users/liggitt/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/liggitt",
    "id": 980082,
    "login": "liggitt",
    "node_id": "MDQ6VXNlcjk4MDA4Mg==",
    "organizations_url": "https://api.github.com/users/liggitt/orgs",
    "received_events_url": "https://api.github.com/users/liggitt/received_events",
    "repos_url": "https://api.github.com/users/liggitt/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/liggitt/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/liggitt/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/liggitt"
  },
  "comments": 1,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/30947/comments",
  "created_at": "2016-08-19T00:41:00Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/30947/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/30947",
  "id": 172031941,
  "labels": [
    {
      "color": "0052cc",
      "default": false,
      "description": null,
      "id": 105152717,
      "name": "area/test",
      "node_id": "MDU6TGFiZWwxMDUxNTI3MTc=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/area/test"
    },
    {
      "color": "f7c6c7",
      "default": false,
      "description": "Categorizes issue or PR as related to a flaky test.",
      "id": 264749912,
      "name": "kind/flake",
      "node_id": "MDU6TGFiZWwyNjQ3NDk5MTI=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/flake"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/30947/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWUxNzIwMzE5NDE=",
  "number": 30947,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "Integration test flake: Data race in k8s.io/kubernetes/pkg/controller/volume/persistentvolume",
  "updated_at": "2016-08-21T18:36:46Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/30947",
  "user": {
    "avatar_url": "https://avatars3.githubusercontent.com/u/5821883?v=4",
    "events_url": "https://api.github.com/users/Random-Liu/events{/privacy}",
    "followers_url": "https://api.github.com/users/Random-Liu/followers",
    "following_url": "https://api.github.com/users/Random-Liu/following{/other_user}",
    "gists_url": "https://api.github.com/users/Random-Liu/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/Random-Liu",
    "id": 5821883,
    "login": "Random-Liu",
    "node_id": "MDQ6VXNlcjU4MjE4ODM=",
    "organizations_url": "https://api.github.com/users/Random-Liu/orgs",
    "received_events_url": "https://api.github.com/users/Random-Liu/received_events",
    "repos_url": "https://api.github.com/users/Random-Liu/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/Random-Liu/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/Random-Liu/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/Random-Liu"
  }
}